using System;
using System.Collections.Generic;

class Program
{
    static void Main(string[] args)
    {

        //1.Create a list of integers. Ask the user for a number to divide each number 
        //  in the list by.Write a loop that takes each integer in the list, 
        //  divides it by the number the user entered, and displays the result to the screen.
        //2.Run that code, entering in non - zero numbers as the user.Look at the displayed results.
        //3.Run that code, entering in zero as the number to divide by.Note any error messages you get.
        //4.Run that code, entering in a string as the number to divide by.Note any error messages you get.
        //5.Now put the loop in a try/catch block.Below and outside of the try/catch block, 
        //   make the program display a message to the display to let you know the program has 
        //   emerged from the try/catch block and continued on with program execution.
        //   In the catch block, display the error message to the screen.
        //   Then try various combinations of user input: valid numbers, zero and a string.
        //   Ensure the proper error messages display on the screen, 
        //   and that the code after the try/catch block gets executed.


        //List<double> num = new List<double>() { 89, 99, 100, 128, 256, 328, 769, 572 };
        //Console.WriteLine("Enter a number: ");
        //int userInput = Convert.ToInt16(Console.ReadLine());
        //for (int i = 0; i < num.Count; i++)
        //{
        //    Console.WriteLine(num[i] / userInput);
        //}
        //    try
        //{
        //    List<double> num = new List<double>() { 89, 99, 100, 128, 256, 328, 769, 572 };
        //    Console.WriteLine("Enter a number: ");
        //    int userInput = Convert.ToInt16(Console.ReadLine());
        //    for (int i = 0; i < num.Count; i++)
        //    {
        //        Console.WriteLine(num[i] / userInput);
        //    }

        //}
        //catch (DivideByZeroException ex)
        //{
        //    Console.WriteLine(ex.Message);
        //    return;
        //}
        //catch (FormatException ex)
        //{
        //    Console.WriteLine(ex.Message);
        //    return;
        //}
        //finally
        //{
        //    Console.WriteLine("This is the End. ");
        //    Console.ReadLine();
        //}
    }
}